File: app/src/main/java/com/hustunique/morii/edit/EditPresenter.java
package com.hustunique.morii.edit;

import android.content.Context;

public class EditPresenter implements EditContract.IPresenter,EditContract.IListener{
    private final EditContract.IView view;
    private final EditContract.IModel model;

    EditPresenter(Context context) {
        this.view = (EditContract.IView) context;
        this.model = new EditModel(this);
        model.setAppCompatActivityUse(context);
    }

    @Override
    public void getPicture(){
        model.getPicture();
    }

    @Override
    public void setIt(String path) {
        view.setAddPhoto(path);
    }
}

File: app/src/main/java/com/hustunique/morii/edit/EditPresenter.kt
package com.hustunique.morii.edit

import com.hustunique.morii.edit.EditContract.IListener
import com.hustunique.morii.edit.EditContract.IModel
import android.content.*

class EditPresenter internal constructor(context: Context) : EditContract.IPresenter, IListener {
    private val view: EditContract.IView
    private val model: IModel
    override val picture: Unit
        get() {
            model.picture
        }

    override fun setIt(path: String) {
        view.setAddPhoto(path)
    }

    init {
        view = context as EditContract.IView
        model = EditModel(this)
        model.setAppCompatActivityUse(context)
    }
}